exporter/elasticsearchexporter/exporter.go

```diff
@@ -28,7 +27,7 @@ type elasticsearchTracesExporter struct {
 	model       mappingModel
 }
 
-func newTracesExporter(logger *zap.Logger, cfg *Config) (*elasticsearchTracesExporter, error) {
+func newExporter(logger *zap.Logger, cfg *Config, index string, dynamicIndex bool) (*elasticsearchExporter, error) {
```

Comment:
[**Non blocking/discussion**] I was thinking of refactoring this a bit more (possibly in the future) to use the same instance of the exporter (and thus the bulk indexer) for both logs and traces (and eventually metrics) - IMO, this will allow us to better reason about resource consumption.

Reply:
I like the idea, but I'm unsure how one would go about it. Is there any prior art? Probably worth opening an issue so this idea and any discussion doesn't get lost.

Reply:
There is [SharedComponent](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/internal/sharedcomponent/sharedcomponent.go) however we can't probably use it as is.

> Probably worth opening an issue so this idea and any discussion doesn't get lost.

Sounds good, I will create an issue.

Reply:
Created an issue: https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/33326

I have a few ideas on how this could look like. Will create a followup PR for discussions after this PR is merged.